xapian-core/backends/flint_lock.cc

```diff
@@ -150,6 +149,23 @@ FlintLock::lock(bool exclusive, bool wait, string & explanation) {
 	errno_to_string(errno, explanation);
 	return ((errno == EMFILE || errno == ENFILE) ? FDLIMIT : UNKNOWN);
     }
+    
+    #if defined  __CYGWIN__ || defined __WIN32__
+    OVERLAPPED xOverlapped;     //Overlapper structure required by LockFileEx()
+    xOverlapped.Offset = 0;     // Low order offset position of file in database
+    xOverlapped.OffsetHigh = 0; //High order offset position of file in database
+    
+    if (!(LockFileEx(hFile, LOCKFILE_EXCLUSIVE_LOCK, LOCKFILE_FAIL_IMMEDIATELY, 0, MAXDWORD, MAXDWORD, &xOverlapped))) {
+                               // Requires Exclusive access hFile rather than standard filename.c_str() from lockfd
+                                 
+     GetLastError();   // Lock Failed check for errors
+     return UNKNOWN;                 
+    }
+    fd = fnm;                  
+    return SUCCESS;          
+
+#endif
+
```

Comment:
You're adding this code to the `#else` branch, so it'll never actually get used, as the first condition checked is:

```
#if defined __CYGWIN__ || defined __WIN32__
```

The code you've added here wouldn't actually compile, which is what tipped me off - `fd = fnm;` can't work as `fd` is an `int` here, while `fnm` is `std::string`.

Your indentation is off too.